sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/PlanAdaptiveDynamicPruningFilters.scala

```diff
@@ -65,7 +70,7 @@ case class PlanAdaptiveDynamicPruningFilters(
           DynamicPruningExpression(InSubqueryExec(value, broadcastValues, exprId))
         } else if (onlyInBroadcast) {
           DynamicPruningExpression(Literal.TrueLiteral)
-        } else {
+        } else if (canBroadcastBySize(buildPlan, conf)) {
```

Comment:
this can be over estimated. The final plan has an `Aggregate` which may dramatically reduce the data size.